compass/ocean/tests/utility/create_salin_restoring/extrap_salin.py

```diff
@@ -0,0 +1,122 @@
+from datetime import datetime
+
+import numpy as np
+import xarray as xr
+from scipy.signal import convolve2d
+
+from compass.step import Step
+
+
+class ExtrapSalin(Step):
+    """
+    Extrapolate WOA 2023 monthly sea surface salinity data into missing ocean
+    regions, including ice cavities and coasts
+
+    Attributes
+    ----------
+    woa_filename : str
+        The name of the output file name after extrapolation
+
+    """
+    def __init__(self, test_case):
+        """
+        Create a new test case
+
+        Parameters
+        ----------
+        test_case : compass.ocean.tests.utility.create_salin_restoring.
+        CreateSalinRestoring
+            The test case this step belongs to
+
+        """
+        super().__init__(test_case=test_case, name='extrap', cpus_per_task=64,
+                         min_cpus_per_task=1, openmp_threads=1)
+
+        self.add_input_file(
+            filename='woa_surface_salinity_monthly.nc',
+            target='../salinity_restoring/woa_surface_salinity_monthly.nc')
+
+        self.woa_filename = None
+
+    def setup(self):
+        """
+        Determine the output filename
+        """
+
+        now = datetime.now()
+
+        datestring = now.strftime("%Y%m%d")
+
+        self.woa_filename = f'woa23_decav_0.25_sss_monthly_extrap.\
+                              {datestring}.nc'
```

Comment:
It isn't okay to put a continuation line in the middle of a string.  Here's how to do that properly:
```suggestion
        self.woa_filename = f'woa23_decav_0.25_sss_monthly_extrap.' \
                            f'{datestring}.nc'
```